enhancements/network/on-prem-dual-stack-vips.md

```diff
@@ -0,0 +1,421 @@
+---
+title: dual-stack-vips
+authors:
+  - @cybertron
+reviewers:
+  - @creydr
+  - @dougsland
+approvers:
+  - @patrickdillon
+  - @kirankt
+  - @shardy
+  - @cgwalters
+  - @kikisdeliveryservice
+  - @danwinship
+api-approvers:
+  - ???
+creation-date: 2022-03-01
+last-updated: 2022-03-24
+tracking-link:
+  - https://issues.redhat.com/browse/SDN-2213
+see-also:
+  - "/enhancements/on-prem-service-load-balancers.md"
+replaces:
+superseded-by:
+---
+
+# Dual Stack VIPs
+
+## Summary
+
+Originally the on-prem loadbalancer architecture supported only single-stack
+ipv4 or ipv6 deployments. When dual stack support was added for the baremetal
+platform, we carried over the VIP configuration that only allowed for a single
+VIP to be specified. The thinking was that in dual stack clusters every node
+would have access to either ipv4 or ipv6 endpoints. While this is true of the
+nodes, it is not necessarily true of the applications that run in or out of
+the cluster and may need to interact with the API or ingress services.
+
+This design is proposing the addition of a second pair of VIPs for dual stack
+clusters so API and ingress are accessible from either ip version.
+
+## Motivation
+
+Customer applications on dual stack clusters may only have ipv6 connectivity,
+but the cluster VIPs will be ipv4 (at this time ipv6 VIPs are not supported
+in dual stack clusters).
+
+### Goals
+
+Create both ipv4 and ipv6 VIPs on dual stack clusters so single stack
+applications have the connectivity they require.
+
+### Non-Goals
+
+Making the `kubernetes.default` Service dual stack is not supported and will
+not be addressed by this work. We are only adding dual stack access to the
+API via the VIPs.
+
+## Proposal
+
+### User Stories
+
+As a deployer of a dual stack OpenShift cluster, I want to run ipv6-only
```

Comment:
Another significant story is hub/spoke mode where the spoke is ipv6 only

Reply:
Is there a meaningful difference between "spoke" and "application"?

Reply:
Different cluster. Different user story. But not siginificant